diff --git a/event.c b/event.c
index dec50f0..8996398 100644
--- a/event.c
+++ b/event.c
@@ -43,6 +43,7 @@ static int parse_beacon_hint_chan(struct nlattr *tb,
 	return 0;
 }
 
+#ifdef IW_FULL
 static void print_frame(struct print_event_args *args, struct nlattr *attr)
 {
 	uint8_t *frame;
@@ -343,7 +344,7 @@ static void parse_vendor_event(struct nlattr **attrs, bool dump)
 out:
 	printf("\n");
 }
-
+#endif
 static void parse_nan_term(struct nlattr **attrs)
 {
 	struct nlattr *func[NL80211_NAN_FUNC_ATTR_MAX + 1];
@@ -913,15 +914,18 @@ static void parse_assoc_comeback(struct nlattr **attrs, int command)
 static int print_event(struct nl_msg *msg, void *arg)
 {
 	struct genlmsghdr *gnlh = nlmsg_data(nlmsg_hdr(msg));
-	struct nlattr *tb[NL80211_ATTR_MAX + 1], *nst;
+	struct nlattr *tb[NL80211_ATTR_MAX + 1];
 	struct print_event_args *args = arg;
 	char ifname[100];
 	char macbuf[6*3];
 	__u8 reg_type;
 	struct ieee80211_beacon_channel chan_before_beacon,  chan_after_beacon;
 	__u32 wiphy_idx = 0;
+#ifdef IW_FULL
+	struct nlattr *nst;
 	int rem_nst;
 	__u16 status;
+#endif
 
 	if (args->time || args->reltime || args->ctime) {
 		unsigned long long usecs, previous;
@@ -973,6 +977,7 @@ static int print_event(struct nl_msg *msg, void *arg)
 	}
 
 	switch (gnlh->cmd) {
+#ifdef IW_FULL
 	case NL80211_CMD_NEW_WIPHY:
 		printf("renamed to %s\n", nla_get_string(tb[NL80211_ATTR_WIPHY_NAME]));
 		break;
@@ -1008,6 +1013,7 @@ static int print_event(struct nl_msg *msg, void *arg)
 	case NL80211_CMD_SCHED_SCAN_RESULTS:
 		printf("got scheduled scan results\n");
 		break;
+#endif
 	case NL80211_CMD_WIPHY_REG_CHANGE:
 	case NL80211_CMD_REG_CHANGE:
 		if (gnlh->cmd == NL80211_CMD_WIPHY_REG_CHANGE)
@@ -1090,6 +1096,7 @@ static int print_event(struct nl_msg *msg, void *arg)
 		mac_addr_n2a(macbuf, nla_data(tb[NL80211_ATTR_MAC]));
 		printf("del station %s\n", macbuf);
 		break;
+#ifdef IW_FULL
 	case NL80211_CMD_JOIN_IBSS:
 		mac_addr_n2a(macbuf, nla_data(tb[NL80211_ATTR_MAC]));
 		printf("IBSS %s joined\n", macbuf);
@@ -1264,6 +1271,7 @@ static int print_event(struct nl_msg *msg, void *arg)
 	case NL80211_CMD_DEL_WIPHY:
 		printf("delete wiphy\n");
 		break;
+#endif
 	case NL80211_CMD_PEER_MEASUREMENT_RESULT:
 		parse_pmsr_result(tb, args);
 		break;
@@ -1298,8 +1306,7 @@ static int print_event(struct nl_msg *msg, void *arg)
 		parse_assoc_comeback(tb, gnlh->cmd);
 		break;
 	default:
-		printf("unknown event %d (%s)\n",
-		       gnlh->cmd, command_name(gnlh->cmd));
+		printf("unknown event %d\n", gnlh->cmd);
 		break;
 	}
 
diff --git a/info.c b/info.c
index c5e863f..49b44c6 100644
--- a/info.c
+++ b/info.c
@@ -19,6 +19,7 @@ static void print_flag(const char *name, int *open)
 	*open = 1;
 }
 
+#ifdef IW_FULL
 static char *cipher_name(__u32 c)
 {
 	static char buf[20];
@@ -56,6 +57,7 @@ static char *cipher_name(__u32 c)
 		return buf;
 	}
 }
+#endif
 
 static int ext_feature_isset(const unsigned char *ext_features, int ext_features_len,
 			     enum nl80211_ext_feature_index ftidx)
@@ -312,19 +314,21 @@ static int print_phy_handler(struct nl_msg *msg, void *arg)
 		[NL80211_FREQUENCY_ATTR_RADAR] = { .type = NLA_FLAG },
 		[NL80211_FREQUENCY_ATTR_MAX_TX_POWER] = { .type = NLA_U32 },
 	};
-
+#ifdef IW_FULL
 	struct nlattr *tb_rate[NL80211_BITRATE_ATTR_MAX + 1];
 	static struct nla_policy rate_policy[NL80211_BITRATE_ATTR_MAX + 1] = {
 		[NL80211_BITRATE_ATTR_RATE] = { .type = NLA_U32 },
 		[NL80211_BITRATE_ATTR_2GHZ_SHORTPREAMBLE] = { .type = NLA_FLAG },
 	};
-
-	struct nlattr *nl_band;
-	struct nlattr *nl_freq;
 	struct nlattr *nl_rate;
 	struct nlattr *nl_cmd;
 	struct nlattr *nl_if, *nl_ftype;
-	int rem_band, rem_freq, rem_rate, rem_cmd, rem_ftype, rem_if;
+	int rem_rate, rem_cmd, rem_ftype, rem_if;
+#endif
+	struct nlattr *nl_band;
+	struct nlattr *nl_freq;
+
+	int rem_band, rem_freq;
 	int open;
 	/*
 	 * static variables only work here, other applications need to use the
@@ -446,6 +450,7 @@ next:
 				}
 			}
 
+#ifdef IW_FULL
 			if (tb_band[NL80211_BAND_ATTR_RATES]) {
 			printf("\t\tBitrates (non-HT):\n");
 			nla_for_each_nested(nl_rate, tb_band[NL80211_BAND_ATTR_RATES], rem_rate) {
@@ -462,6 +467,7 @@ next:
 				printf("\n");
 			}
 			}
+#endif
 		}
 	}
 
@@ -527,6 +533,7 @@ next:
 		printf("\tCoverage class: %d (up to %dm)\n", coverage, 450 * coverage);
 	}
 
+#ifdef IW_FULL
 	if (tb_msg[NL80211_ATTR_CIPHER_SUITES]) {
 		int num = nla_len(tb_msg[NL80211_ATTR_CIPHER_SUITES]) / sizeof(__u32);
 		int i;
@@ -538,6 +545,7 @@ next:
 					cipher_name(ciphers[i]));
 		}
 	}
+#endif
 
 	if (tb_msg[NL80211_ATTR_WIPHY_ANTENNA_AVAIL_TX] &&
 	    tb_msg[NL80211_ATTR_WIPHY_ANTENNA_AVAIL_RX])
@@ -555,9 +563,11 @@ next:
 		print_iftype_list("\tSupported interface modes", "\t\t",
 				  tb_msg[NL80211_ATTR_SUPPORTED_IFTYPES]);
 
+#ifdef IW_FULL
 	if (tb_msg[NL80211_ATTR_SOFTWARE_IFTYPES])
 		print_iftype_list("\tsoftware interface modes (can always be added)",
 				  "\t\t", tb_msg[NL80211_ATTR_SOFTWARE_IFTYPES]);
+#endif
 
 	if (tb_msg[NL80211_ATTR_INTERFACE_COMBINATIONS]) {
 		struct nlattr *nl_combi;
@@ -647,6 +657,7 @@ broken_combination:
 			printf("\tinterface combinations are not supported\n");
 	}
 
+#ifdef IW_FULL
 	if (tb_msg[NL80211_ATTR_SUPPORTED_COMMANDS]) {
 		printf("\tSupported commands:\n");
 		nla_for_each_nested(nl_cmd, tb_msg[NL80211_ATTR_SUPPORTED_COMMANDS], rem_cmd)
@@ -744,6 +755,7 @@ broken_combination:
 				printf("\t\t * wake up on TCP connection\n");
 		}
 	}
+#endif
 
 	if (tb_msg[NL80211_ATTR_ROAM_SUPPORT])
 		printf("\tDevice supports roaming.\n");
@@ -782,6 +794,7 @@ broken_combination:
 		}
 	}
 
+#ifdef IW_FULL
 	if (tb_msg[NL80211_ATTR_FEATURE_FLAGS]) {
 		unsigned int features = nla_get_u32(tb_msg[NL80211_ATTR_FEATURE_FLAGS]);
 
@@ -849,6 +862,10 @@ broken_combination:
 
 	if (tb_msg[NL80211_ATTR_TDLS_SUPPORT])
 		printf("\tDevice supports T-DLS.\n");
+		if (features & NL80211_FEATURE_TDLS_CHANNEL_SWITCH)
+			printf("\tDevice supports TDLS channel switching\n");
+	}
+#endif
 
 	if (tb_msg[NL80211_ATTR_EXT_FEATURES]) {
 		struct nlattr *tb = tb_msg[NL80211_ATTR_EXT_FEATURES];
@@ -914,6 +931,7 @@ TOPLEVEL(list, NULL, NL80211_CMD_GET_WIPHY, NLM_F_DUMP, CIB_NONE, handle_info,
 	 "List all wireless devices and their capabilities.");
 TOPLEVEL(phy, NULL, NL80211_CMD_GET_WIPHY, NLM_F_DUMP, CIB_NONE, handle_info, NULL);
 
+#ifdef IW_FULL
 static int handle_commands(struct nl80211_state *state, struct nl_msg *msg,
 			   int argc, char **argv, enum id_input id)
 {
@@ -925,6 +943,7 @@ static int handle_commands(struct nl80211_state *state, struct nl_msg *msg,
 }
 TOPLEVEL(commands, NULL, NL80211_CMD_GET_WIPHY, 0, CIB_NONE, handle_commands,
 	 "list all known commands and their decimal & hex value");
+#endif
 
 static int print_feature_handler(struct nl_msg *msg, void *arg)
 {
diff --git a/scan.c b/scan.c
index faf406d..9857561 100644
--- a/scan.c
+++ b/scan.c
@@ -593,6 +593,7 @@ static void print_supprates(const uint8_t type, uint8_t len,
 	printf("\n");
 }
 
+#ifdef IW_FULL
 static void print_rm_enabled_capabilities(const uint8_t type, uint8_t len,
 			    const uint8_t *data,
 			    const struct print_ies_data *ie_buffer)
@@ -757,6 +758,7 @@ static void print_ap_channel_report(const uint8_t type, uint8_t len, const uint8
 	}
 	printf("\n");
 }
+#endif
 
 static void print_cipher(const uint8_t *data)
 {
@@ -1099,6 +1101,7 @@ static void print_ht_capa(const uint8_t type, uint8_t len, const uint8_t *data,
 	print_ht_mcs(data + 3);
 }
 
+#ifdef IW_FULL
 static const char* ntype_11u(uint8_t t)
 {
 	switch (t) {
@@ -1281,6 +1284,7 @@ static void print_tx_power_envelope(const uint8_t type, uint8_t len,
 			printf("\t\t * %s: %i dBm\n", power_names[i], power_val);
 	}
 }
+#endif
 
 static const char *ht_secondary_offset[4] = {
 	"no secondary",
@@ -1308,6 +1312,7 @@ static void print_ht_op(const uint8_t type, uint8_t len, const uint8_t *data,
 	printf("\t\t * secondary channel offset: %s\n",
 		ht_secondary_offset[data[1] & 0x3]);
 	printf("\t\t * STA channel width: %s\n", sta_chan_width[(data[1] & 0x4)>>2]);
+	return;
 	printf("\t\t * RIFS: %d\n", (data[1] & 0x8)>>3);
 	printf("\t\t * HT protection: %s\n", protection[data[2] & 0x3]);
 	printf("\t\t * non-GF present: %d\n", (data[2] & 0x4) >> 2);
@@ -1320,6 +1325,7 @@ static void print_ht_op(const uint8_t type, uint8_t len, const uint8_t *data,
 	printf("\t\t * PCO phase: %d\n", (data[5] & 0x8) >> 3);
 }
 
+#ifdef IW_FULL
 static void print_capabilities(const uint8_t type, uint8_t len,
 			       const uint8_t *data,
 			       const struct print_ies_data *ie_buffer)
@@ -1484,6 +1490,7 @@ static void print_capabilities(const uint8_t type, uint8_t len,
 
 	printf("\n");
 }
+#endif
 
 static void print_tim(const uint8_t type, uint8_t len, const uint8_t *data,
 		      const struct print_ies_data *ie_buffer)
@@ -1496,11 +1503,13 @@ static void print_tim(const uint8_t type, uint8_t len, const uint8_t *data,
 	printf("\n");
 }
 
+#ifdef IW_FULL
 static void print_ibssatim(const uint8_t type, uint8_t len, const uint8_t *data,
 			   const struct print_ies_data *ie_buffer)
 {
 	printf(" %d TUs\n", (data[1] << 8) + data[0]);
 }
+#endif
 
 static void print_vht_capa(const uint8_t type, uint8_t len, const uint8_t *data,
 			   const struct print_ies_data *ie_buffer)
@@ -1563,6 +1572,7 @@ static void print_supp_op_classes(const uint8_t type, uint8_t len,
 		}
 }
 
+#ifdef IW_FULL
 static void print_measurement_pilot_tx(const uint8_t type, uint8_t len,
 				       const uint8_t *data,
 				       const struct print_ies_data *ie_buffer)
@@ -1626,6 +1636,7 @@ static void print_obss_scan_params(const uint8_t type, uint8_t len,
 	printf("\t\t * OBSS Scan Activity Threshold: %d.%02d %%\n",
 		((data[13] << 8) | data[12]) / 100, ((data[13] << 8) | data[12]) % 100);
 }
+#endif
 
 static void print_secchan_offs(const uint8_t type, uint8_t len,
 			       const uint8_t *data,
@@ -1637,6 +1648,7 @@ static void print_secchan_offs(const uint8_t type, uint8_t len,
 		printf(" %d\n", data[0]);
 }
 
+#ifdef IW_FULL
 static void print_bss_load(const uint8_t type, uint8_t len, const uint8_t *data,
 			   const struct print_ies_data *ie_buffer)
 {
@@ -1773,6 +1785,7 @@ static void print_s1g_oper(const uint8_t type, uint8_t len,
 	printf("\t\t\tFor 3 SS: %s\n", s1g_ss_min_support(data[5] & 0x3));
 	printf("\t\t\tFor 4 SS: %s\n", s1g_ss_min_support((data[5] >> 4) & 0x3));
 }
+#endif
 
 struct ie_print {
 	const char *name;
@@ -1819,26 +1832,27 @@ static const struct ie_print ieprinters[] = {
 	[0] = { "SSID", print_ssid, 0, 32,
 		 BIT(PRINT_SCAN) | BIT(PRINT_LINK) | BIT(PRINT_LINK_MLO_MLD), },
 	[1] = { "Supported rates", print_supprates, 0, 255, BIT(PRINT_SCAN), },
-	[3] = { "DS Parameter set", print_ds, 1, 1, BIT(PRINT_SCAN), },
+	[45] = { "HT capabilities", print_ht_capa, 26, 26, BIT(PRINT_SCAN), },
+	[48] = { "RSN", print_rsn, 2, 255, BIT(PRINT_SCAN), },
+	[50] = { "Extended supported rates", print_supprates, 0, 255, BIT(PRINT_SCAN), },
+	[59] = { "Supported operating classes", print_supp_op_classes, 1, 255, BIT(PRINT_SCAN), },
+	[61] = { "HT operation", print_ht_op, 22, 22, BIT(PRINT_SCAN), },
+	[62] = { "Secondary Channel Offset", print_secchan_offs, 1, 1, BIT(PRINT_SCAN), },
+	[191] = { "VHT capabilities", print_vht_capa, 12, 255, BIT(PRINT_SCAN), },
+	[192] = { "VHT operation", print_vht_oper, 5, 255, BIT(PRINT_SCAN), },
 	[5] = { "TIM", print_tim, 4, 255, BIT(PRINT_SCAN), },
+#ifdef IW_FULL
+	[3] = { "DS Parameter set", print_ds, 1, 1, BIT(PRINT_SCAN), },
 	[6] = { "IBSS ATIM window", print_ibssatim, 2, 2, BIT(PRINT_SCAN), },
 	[7] = { "Country", print_country, 3, 255, BIT(PRINT_SCAN), },
 	[11] = { "BSS Load", print_bss_load, 5, 5, BIT(PRINT_SCAN), },
 	[32] = { "Power constraint", print_powerconstraint, 1, 1, BIT(PRINT_SCAN), },
 	[35] = { "TPC report", print_tpcreport, 2, 2, BIT(PRINT_SCAN), },
 	[42] = { "ERP", print_erp, 1, 255, BIT(PRINT_SCAN), },
-	[45] = { "HT capabilities", print_ht_capa, 26, 26, BIT(PRINT_SCAN), },
 	[47] = { "ERP D4.0", print_erp, 1, 255, BIT(PRINT_SCAN), },
 	[51] = { "AP Channel Report", print_ap_channel_report, 1, 255, BIT(PRINT_SCAN), },
-	[59] = { "Supported operating classes", print_supp_op_classes, 1, 255, BIT(PRINT_SCAN), },
 	[66] = { "Measurement Pilot Transmission", print_measurement_pilot_tx, 1, 255, BIT(PRINT_SCAN), },
 	[74] = { "Overlapping BSS scan params", print_obss_scan_params, 14, 255, BIT(PRINT_SCAN), },
-	[61] = { "HT operation", print_ht_op, 22, 22, BIT(PRINT_SCAN), },
-	[62] = { "Secondary Channel Offset", print_secchan_offs, 1, 1, BIT(PRINT_SCAN), },
-	[191] = { "VHT capabilities", print_vht_capa, 12, 255, BIT(PRINT_SCAN), },
-	[192] = { "VHT operation", print_vht_oper, 5, 255, BIT(PRINT_SCAN), },
-	[48] = { "RSN", print_rsn, 2, 255, BIT(PRINT_SCAN), },
-	[50] = { "Extended supported rates", print_supprates, 0, 255, BIT(PRINT_SCAN), },
 	[70] = { "RM enabled capabilities", print_rm_enabled_capabilities, 5, 5, BIT(PRINT_SCAN), },
 	[113] = { "MESH Configuration", print_mesh_conf, 7, 7, BIT(PRINT_SCAN), },
 	[114] = { "MESH ID", print_ssid, 0, 32, BIT(PRINT_SCAN) | BIT(PRINT_LINK), },
@@ -1850,6 +1864,7 @@ static const struct ie_print ieprinters[] = {
 	[214] = { "Short beacon interval", print_short_beacon_int, 2, 2, BIT(PRINT_SCAN), },
 	[217] = { "S1G capabilities", print_s1g_capa, 15, 15, BIT(PRINT_SCAN), },
 	[232] = { "S1G operation", print_s1g_oper, 6, 6, BIT(PRINT_SCAN), },
+#endif
 };
 
 static void print_wifi_wpa(const uint8_t type, uint8_t len, const uint8_t *data,
@@ -1865,6 +1880,7 @@ static void print_wifi_osen(const uint8_t type, uint8_t len,
 	print_osen_ie("OSEN", "OSEN", len, data);
 }
 
+#ifdef IW_FULL
 static bool print_wifi_wmm_param(const uint8_t *data, uint8_t len)
 {
 	int i;
@@ -2182,11 +2198,14 @@ static void print_wifi_wps(const uint8_t type, uint8_t len, const uint8_t *data,
 		printf("\n");
 	}
 }
+#endif
 
 static const struct ie_print wifiprinters[] = {
 	[1] = { "WPA", print_wifi_wpa, 2, 255, BIT(PRINT_SCAN), },
+#ifdef IW_FULL
 	[2] = { "WMM", print_wifi_wmm, 1, 255, BIT(PRINT_SCAN), },
 	[4] = { "WPS", print_wifi_wps, 0, 255, BIT(PRINT_SCAN), },
+#endif
 };
 
 static inline void print_p2p(const uint8_t type, uint8_t len,
@@ -2283,6 +2302,7 @@ static inline void print_hs20_ind(const uint8_t type, uint8_t len,
 		printf("\t\tUnexpected length: %i\n", len);
 }
 
+#ifdef IW_FULL
 static void print_wifi_owe_tarns(const uint8_t type, uint8_t len,
 				 const uint8_t *data,
 				 const struct print_ies_data *ie_buffer)
@@ -2310,12 +2330,15 @@ static void print_wifi_owe_tarns(const uint8_t type, uint8_t len,
 		printf("\t\tChannel Info: %u\n", data[ssid_len + 8]);
 	}
 }
+#endif
 
 static const struct ie_print wfa_printers[] = {
 	[9] = { "P2P", print_p2p, 2, 255, BIT(PRINT_SCAN), },
 	[16] = { "HotSpot 2.0 Indication", print_hs20_ind, 1, 255, BIT(PRINT_SCAN), },
 	[18] = { "HotSpot 2.0 OSEN", print_wifi_osen, 1, 255, BIT(PRINT_SCAN), },
+#ifdef IW_FULL
 	[28] = { "OWE Transition Mode", print_wifi_owe_tarns, 7, 255, BIT(PRINT_SCAN), },
+#endif
 };
 
 static void print_vendor(unsigned char len, unsigned char *data,
@@ -2436,6 +2459,7 @@ void print_ies(unsigned char *ie, int ielen, bool unknown,
 			print_vendor(ie[1], ie + 2, unknown, ptype);
 		} else if (ie[0] == 255 /* extension */) {
 			print_extension(ie[1], ie + 2, unknown, ptype);
+#ifdef IW_FULL
 		} else if (unknown) {
 			int i;
 
@@ -2443,6 +2467,7 @@ void print_ies(unsigned char *ie, int ielen, bool unknown,
 			for (i=0; i<ie[1]; i++)
 				printf(" %.2x", ie[2+i]);
 			printf("\n");
+#endif
 		}
 		ielen -= ie[1] + 2;
 		ie += ie[1] + 2;
@@ -2483,6 +2508,7 @@ static void print_capa_non_dmg(__u16 capa)
 		printf(" ESS");
 	if (capa & WLAN_CAPABILITY_IBSS)
 		printf(" IBSS");
+#ifdef IW_FULL
 	if (capa & WLAN_CAPABILITY_CF_POLLABLE)
 		printf(" CfPollable");
 	if (capa & WLAN_CAPABILITY_CF_POLL_REQUEST)
@@ -2511,6 +2537,7 @@ static void print_capa_non_dmg(__u16 capa)
 		printf(" DelayedBACK");
 	if (capa & WLAN_CAPABILITY_IMM_BACK)
 		printf(" ImmediateBACK");
+#endif
 }
 
 static int print_bss_handler(struct nl_msg *msg, void *arg)
@@ -2601,8 +2628,10 @@ static int print_bss_handler(struct nl_msg *msg, void *arg)
 		else
 			printf("\tfreq: %d\n", freq);
 
+#ifdef IW_FULL
 		if (freq > 45000)
 			is_dmg = true;
+#endif
 	}
 	if (bss[NL80211_BSS_BEACON_INTERVAL])
 		printf("\tbeacon interval: %d TUs\n",
diff --git a/util.c b/util.c
index d36dbdc..68419c6 100644
--- a/util.c
+++ b/util.c
@@ -147,6 +147,7 @@ const char *iftype_name(enum nl80211_iftype iftype)
 	return modebuf;
 }
 
+#ifdef IW_FULL
 static const char *commands[NL80211_CMD_MAX + 1] = {
 #include "nl80211-commands.inc"
 };
@@ -160,6 +161,7 @@ const char *command_name(enum nl80211_commands cmd)
 	sprintf(cmdbuf, "Unknown command (%d)", cmd);
 	return cmdbuf;
 }
+#endif
 
 int ieee80211_channel_to_frequency(int chan, enum nl80211_band band)
 {
